src/main/java/ch/njol/skript/SkriptEventHandler.java

```diff
@@ -251,35 +249,40 @@ public static void registerBukkitEvent(Trigger trigger, Class<? extends Event> e
 	 * @param trigger The Trigger to unregister events for.
 	 */
 	public static void unregisterBukkitEvents(Trigger trigger) {
-		triggers.removeIf(pair -> {
-			if (pair.getSecond() != trigger)
-				return false;
+		Iterator<Map.Entry<Class<? extends Event>, Trigger>> entryIterator = triggers.entries().iterator();
+		Map.Entry<Class<? extends Event>, Trigger> entry;
+		entryLoop: while (entryIterator.hasNext()) {
+			entry = entryIterator.next();
+			if (entry.getValue() != trigger)
```

Comment:
```suggestion
			if (!entry.getValue().equal(trigger))
```

Reply:
I think a reference comparison is fine here. There shouldn't be two different triggers representing the same thing.

Reply:
That's an assumption versus an exact match. == is for exact matching which is fine in Spigot aside from locations/vectors which is why developers are so conformable with it. Should be handling matching against with equals method.

Reply:
Exact matching is all we need - there's no reason to call the equals method when the operator will suffice